.github/workflows/core-contracts-storage-check.yml

```diff
@@ -0,0 +1,55 @@
+name: Check For Core Contracts Storage Changes
+
+on:
+  push:
+    branches:
+      - development
+  pull_request:
+
+jobs:
+  provide_contracts:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - id: set-matrix
+        working-directory: packages/contracts
+        run: |
+          forge tree | grep -E '^src/dollar/core' | cut -d' ' -f1 | xargs basename -s | cut -d'.' -f1 | xargs -I{} echo src/dollar/core/{}.sol:{} >> contracts.txt
+          echo "::set-output name=matrix::$(cat contracts.txt | jq -R -s -c 'split("\n")[:-1]')"
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  check_storage_layout:
+    needs: provide_contracts
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        contract: ${{ fromJSON(needs.provide_contracts.outputs.matrix) }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+        
+      - name: Check For Core Contracts Storage Changes
+        uses: Rubilmax/foundry-storage-check@main
```

Comment:
can you explain why `Rubilmax/foundry-storage-check@main` it's pointing to a specific branch `main`, instead of a version?. This task will run on PR's it's best to have a stable version

Reply:
I think it's ok to use the `main` branch. I doubt that https://github.com/Rubilmax/foundry-storage-check will be updated frequently + we need to check for the latest action's release from time to time if we use a specific tag.

Reply:
the original repo sample shows v3.6. I would use v3.6 i think there is no issue for us in keeping and eye on update.

But i am also in favor of what you say